lightshow/database.py

```diff
@@ -161,18 +161,16 @@ def from_materials_project(cls, **kwargs):
         except KeyError:
             pass
 
-        mpr = MPRester(api_key)
-        method = _get_method(kwargs.get("method"), mpr=mpr)
-
-        try:
-            kwargs.pop("method")
-        except KeyError:
-            pass
-
-        if method is not None:
-            searched = getattr(mpr.mateirals, method).search(**kwargs)
-        else:
-            searched = mpr.materials.search(**kwargs)
+        with MPRester(api_key) as mpr:
+            method = _get_method(kwargs.get("method"), mpr=mpr)
+            try:
+                kwargs.pop("method")
+            else:
```

Comment:
This should be `except KeyError` not `else`.

Reply:
solved